hikari/internal/enums.py

```diff
@@ -219,18 +210,568 @@ def __repr__(cls) -> str:
+        return (self & other) == self
+
+    def none(self: _T, *flags: _T) -> bool:
+        """Check if none of the given flags are part of this value.
+
+        !!! note
+            This is essentially the opposite of `Flag.any`.
+
+        Returns
+        -------
+        builtins.bool
+            `builtins.True` if none of the given flags are part of this value.
+            Otherwise, return `builtins.False`.
+        """
+        return not self.any(*flags)
+
+    def split(self: _T) -> typing.Sequence[_T]:
+        """Return a list of all defined atomic values for this flag.
+
+        Any unrecognised bits will be omitted for brevity.
+
+        The result will be a name-sorted `typing.Sequence` of each membe
+        """
+        return sorted(
+            (member for member in self.__class__._powers_of_2_to_member_map_.values() if member.value & self),
+            # Assumption: powers of 2 already have a cached value.
+            key=lambda m: m._name_,
+        )
+
+    def symmetric_difference(self: _T, other: typing.Union[_T, int]) -> _T:
+        """Return a set with the symmetric differences of two flag sets.
+
+        Equivalent to using the "XOR" `^` operator.
 
+        For `a ^ b`, this can be considered the same as `(a - b) | (b - a)`.
+        """
+        return self.__class__(self._value_ ^ int(other))
 
-# We have to use this fallback, or Pdoc will fail to document some stuff correctly...
-if os.getenv("PDOC3_GENERATING") == "1":  # pragma: no cover
-    from enum import Enum  # noqa: F811 - Redefinition intended
+    def union(self: _T, other: typing.Union[_T, int]) -> _T:
+        """Return a combination of all flags in this set and the other set.
+
+        Equivalent to using the "OR" `~` operator.
+        """
+        return self.__class__(self._value_ | int(other))
+
+    isdisjoint = is_disjoint
+    issubset = is_subset
+    issuperset = is_superset
+    # Exists since Python's `set` type is inconsistent with naming, so this
+    # will prevent tripping people up unnecessarily because we do not
+    # name inconsistently.
+
+    def __bool__(self) -> bool:
+        return bool(self._value_)
+
+    def __int__(self) -> int:
+        return self._value_
+
+    def __iter__(self: _T) -> typing.Iterator[_T]:
+        return iter(self.split())
+
+    def __len__(self) -> int:
+        return len(self.split())
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}.{self.name}: {self.value!r}>"
+
+    def __rsub__(self: _T, other: typing.Union[int, _T]) -> _T:
+        # This logic has to be reversed to be correct, since order matters for
+        # a subtraction operator. This also ensures `int - _T -> _T` is a valid
+        # case for us.
+        if not isinstance(other, self.__class__):
+            other = self.__class__(other)
+        return other - self
+
+    def __str__(self) -> str:
+        return self.name or "NO_NAME"
```

Comment:
Would the `or "NO_NAME"` case here ever be hit? from what I can see you seem to be resolving unknown flags to a generic name anyway

Reply:
true

Reply:
removed it :)